src/libcore/tests/fmt/num.rs

```diff
@@ -38,6 +38,16 @@ fn test_format_int() {
     assert_eq!(format!("{:o}", 1i16), "1");
     assert_eq!(format!("{:o}", 1i32), "1");
     assert_eq!(format!("{:o}", 1i64), "1");
+    assert_eq!(format!("{:e}", 1isize), "1e0");
+    assert_eq!(format!("{:e}", 1i8), "1e0");
+    assert_eq!(format!("{:e}", 1i16), "1e0");
+    assert_eq!(format!("{:e}", 1i32), "1e0");
+    assert_eq!(format!("{:e}", 1i64), "1e0");
+    assert_eq!(format!("{:E}", 1isize), "1E0");
+    assert_eq!(format!("{:E}", 1i8), "1E0");
+    assert_eq!(format!("{:E}", 1i16), "1E0");
+    assert_eq!(format!("{:E}", 1i32), "1E0");
+    assert_eq!(format!("{:E}", 1i64), "1E0");
```

Comment:
please add test cases involving:

* i32::MIN / i64::MIN / i128::MIN

Can (should) the new code handle precision flags?

```rust
assert_eq!(format!("{:.3e}", 123456789), "1.235e9");
```

Reply:
> Can (should) the new code handle precision flags?

It can 't as of 6b12eb9, but it definitely should. I completely overlooked this, thinking I could rely on `pad_integral` to deal with formatting flags.